Fix wxTextCtrl creation in wxGenericListCtrl::EditLabel()

AliKet opened the pull request:
Ensure the edit control is created before sending `wxEVT_LIST_BEGIN_LABEL_EDIT` event, as calling `wxListCtrl::GetEditControl()` from inside the handler should always return a valid pointer

Commit:
Fix wxTextCtrl creation in wxGenericListCtrl::EditLabel()

Ensure the edit control is created before sending wxEVT_LIST_BEGIN_LABEL_EDIT
event, as calling wxListCtrl::GetEditControl() from inside the handler should
always return a valid pointer

vadz commented on src/generic/listctrl.cpp:
```diff
@@ -2391,6 +2391,9 @@ wxTextCtrl *wxListMainWindow::EditLabel(long item, wxClassInfo* textControlClass
     wxASSERT_MSG( textControlClass->IsKindOf(wxCLASSINFO(wxTextCtrl)),
                  wxT("EditLabel() needs a text control") );
 
+    wxTextCtrl * const text = (wxTextCtrl *)textControlClass->CreateObject();
+    m_textctrlWrapper = new wxListTextCtrlWrapper(this, text, item);
+
```
Can we move these lines immediately before `ProcessEvent()` call? This would avoid the problem with orphan text control being left if `wxCHECK_MSG` below fails and would also make it more clear that we need to create the control before calling the handler.

vadz reviewed (commented):
I wonder if we could create the text control on demand in `GetEditControl()` instead? The advantage of doing it like this would be that we don't do it needlessly if editing is vetoed, as I'm a bit concerned that creating the control and immediately destroying it could result in some flicker on the screen.

AliKet commented:
> I wonder if we could create the text control on demand in `GetEditControl()` instead? The advantage of doing it like this would be that we don't do it needlessly if editing is vetoed, as I'm a bit concerned that creating the control and immediately destroying it could result in some flicker on the screen.

In fact, I just followed the comment in `EditLabel()` here: https://github.com/wxWidgets/wxWidgets/blob/32041aa2797c888e60eac52f783193f5cd8175f7/src/msw/listctrl.cpp#L1762-L1768
which shows good reason for why the textctrl is created first before generating the  wxEVT_LIST_BEGIN_LABEL_EDIT event.

AliKet committed:
Apply review suggestion

Commit:
Document that wxLC_EDIT_LABELS flag is required for portable EditLabel()

Since wxListCtrl under wxMSW cannot edit labels without the presence of this flag
and an assertion failure is triggered if the flag is missing, the other ports will
also trigger the assertion if the flag is also missing.

AliKet commented:
The failures are expected now and this PR is dependent on this #23022 for the tests to pass.

AliKet commented:
This should be ready now.

vadz commented:
Thanks, I've merged this now but as 2 commits as I wanted to get rid of the second one -- and also slightly update the last one.
